src/backends/postgresql/statement.cpp

```diff
@@ -781,19 +781,44 @@ void postgresql_statement_backend::describe_column(int colNum, data_type & type,
 
     default:
     {
-        int form = PQfformat(result_, pos);
-        int size = PQfsize(result_, pos);
-        if (form == 0 && size == -1)
+        auto typeCategoryIt = categoryByColumnOID_.find(typeOid);
+        if( typeCategoryIt == categoryByColumnOID_.end() ) 
         {
-            type = dt_string;
+            std::stringstream query;
+            query << "SELECT typcategory FROM pg_type WHERE oid=" << (int)typeOid;
+
+            PGresult* res = PQexec(session_.conn_, query.str().c_str());
```

Comment:
You should use `soci::postgresql_result` here to ensure that `PQclear()` is called. The best proof of its advantage is that you forgot to do it here, which explains the tests failures, I think (I didn't look in the details, but it seems likely).

Reply:
sure